cpp/src/arrow/record_batch.cc

```diff
@@ -78,6 +79,52 @@ class SimpleRecordBatch : public RecordBatch {
 
   std::shared_ptr<ArrayData> column_data(int i) const override { return columns_[i]; }
 
+  Status AddColumn(int i, const std::shared_ptr<Field>& field,
+                   const std::shared_ptr<ArrayData>& column,
+                   std::shared_ptr<RecordBatch>* out) const override {
+    if (i < 0 || i > num_columns() + 1) {
+      return Status::Invalid("Invalid column index");
+    }
+    if (field == nullptr) {
+      std::stringstream ss;
+      ss << "Field " << i << " was null";
+      return Status::Invalid(ss.str());
+    }
+    if (column == nullptr) {
+      std::stringstream ss;
+      ss << "Column " << i << " was null";
+      return Status::Invalid(ss.str());
+    }
```

Comment:
I think these should both be `DCHECK`, since null would indicate a problem with application logic, so should be a "can't fail"

Reply:
I took a look at `SimpleTable::AddColumn`; there `col` is being null-checked -- I think that should also be a DCHECK